core/jobs/types/firebase_adapters.py

```
@@ -0,0 +1,132 @@
+        developer convenience. The password is generated for developers based on
+        the email being signed-up with, and will always be: `md5(email)`.
+        See: `core/templates/services/auth.service.ts`.
+
+        When the EMULATOR receives an email/password sign-up request, it uses
+        "fake" hashing and salting on the password's raw value. We are copying
+        its strategy here to ensure that the import stays consistent with the
+        FRONTEND approach. This is NOT a problem in PROD, because we use SSO
+        (like Google Sign-In) to register there.
+        See: `node_modules/firebase-tools/lib/emulator/auth/operations.js`.
+        """
+        raw_password = hashlib.md5(self.email.encode()).hexdigest()
```

Ruff (0.15.0)

[error] Probable use of insecure hash functions in `hashlib`: `md5`

(S324)